File: 1-two-sum/two-sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    *returnSize=2;
    int * res;
    res=malloc(sizeof(int)*2);
    for(int i=0;i<numsSize;i++){
        for(int j=i+1;j<numsSize;j++){
            int sum=nums[i]+nums[j];
            if(sum==target){
                res[0]=i;
                res[1]=j;
                return res;
            }
        }
    }

    return -1;
}

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int i =0;
        while(i<nums.length){
            if(!map.containsKey(target - nums[i])){
                map.put(nums[i],i);
            }else{
                return new int[]{i,map.get(target - nums[i])};
            }
            i++;
        }

        return new int[]{-1};

    }
}